backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -534,9 +534,11 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
         }
 
         override fun genContinue(destination: IrContinue) {
-            if (destination.loop == loop)
+            if (destination.loop == loop) {
+                if (context.memoryModel == MemoryModel.EXPERIMENTAL)
+                    call(context.llvm.Kotlin_mm_safePointWhileLoopContinue, emptyList())
```

Comment:
Have you considered placing safe points to the beginning of `loopCheck`, to cover both cases -- with and without `continue`? I guess there are pros and cons, so we should at least compare both options.

Reply:
This would make safe point fire even if the loop condition is false at the start. However, we could place safe point at the beginning of loop body instead.

Reply:
Indeed. This would probably be the most efficient solution in terms of code size. Runtime performance is more complicated though.

Reply:
Okay, done.